image/image.go

```diff
@@ -49,6 +49,13 @@ var validRefMediaTypes = []string{
 }
 
 func validate(w walker, refs []string, out *log.Logger) error {
+	var tmperr error
```

Comment:
IMHO, the regular expression check should be at the beginning of the function. As for your recommendation, let the check below line `:=` for `err` to avoid one more variable. But I think it is a little bit of breaking the logic with handling image validation, that means, the original func `validate` is for image validation, and not for parameter input check. So, I think the parameter check should be at the beginning, and the rest of code lines does its original operation, no insertion of other operation, even just only one line.